skyportal/utils/catalog.py

```diff
@@ -0,0 +1,374 @@
+                # Say, keep only Dec>-40 deg to be conservative
+                if dec < -40.0:
+                    continue
+                coords = SkyCoord(ra=float(ra) * u.deg, dec=float(dec) * u.deg)
+                coords_arr.append((coords.ra.deg, coords.dec.deg))
+            except ValueError:
+                print("Problems with the galaxy coordinates?")
+                continue
+
+        # Correct the minimum number of detections
+        ndethist_min_corrected = int(ndethist_min - 1)
+
+        # Correct the jd_trigger if the user specifies to query
+        # also before the trigger
+        if after_trigger is False:
+            jd_trigger = 0
+        q = {
+            "query_type": "cone_search",
+            "query": {
+                "object_coordinates": {
+                    "radec": f"{coords_arr}",
+                    "cone_search_radius": f"{radius}",
+                    "cone_search_unit": "arcmin",
+                },
+                "catalogs": {
+                    "ZTF_alerts": {
+                        "filter": {
+                            "candidate.jd": {'$gt': jd_trigger},
+                            "candidate.drb": {'$gt': 0.8},
+                            "candidate.ndethist": {'$gt': ndethist_min_corrected},
+                            "candidate.jdstarthist": {
+                                '$gt': jd_trigger,
+                                '$lt': jd_trigger + within_days,
+                            },
+                        },
+                        "projection": {
+                            "objectId": 1,
+                            "candidate.rcid": 1,
+                            "candidate.ra": 1,
+                            "candidate.dec": 1,
+                            "candidate.jd": 1,
+                            "candidate.ndethist": 1,
+                            "candidate.jdstarthist": 1,
+                            "candidate.jdendhist": 1,
+                            "candidate.jdendhist": 1,
+                            "candidate.magpsf": 1,
+                            "candidate.sigmapsf": 1,
+                            "candidate.fid": 1,
+                            "candidate.programid": 1,
+                            "candidate.isdiffpos": 1,
+                            "candidate.ndethist": 1,
+                            "candidate.ssdistnr": 1,
+                            "candidate.rb": 1,
+                            "candidate.drb": 1,
+                            "candidate.distpsnr1": 1,
+                            "candidate.sgscore1": 1,
+                            "candidate.srmag1": 1,
+                            "candidate.distpsnr2": 1,
+                            "candidate.sgscore2": 1,
+                            "candidate.srmag2": 1,
+                            "candidate.distpsnr3": 1,
+                            "candidate.sgscore3": 1,
+                            "candidate.srmag3": 1,
```

Comment:
Are all these `1`s just defaults to be edited later?

Reply:
This actually is mongo's way of indicating to return a field or not.